BasicSample/Assets/ARAnchor/Scripts/AnchorsSample.cs

```diff
@@ -78,21 +85,33 @@ public void AnchorsChanged(ARAnchorsChangedEventArgs eventArgs)
                 // AnchorsChanged update. These double adds are ignored, but other added anchors are processed.
                 if (m_anchors.Contains(added)) continue;
 #endif
-                Debug.Log($"Anchor added from changed event: {added.trackableId}, OpenXR Handle: {added.GetOpenXRHandle()}");
+                Debug.Log($"Anchor added: {added.trackableId}, OpenXR Handle: {added.GetOpenXRHandle()}");
                 m_anchors.Add(added);
 
                 // If this anchor being added was requested from the anchor store, it is recognized here
-                if (m_incomingPersistedAnchors.ContainsKey(added.trackableId))
+                if (m_incomingPersistedAnchors.TryGetValue(added.trackableId, out string name))
                 {
-                    added.GetComponent<SampleAnchor>().Name = m_incomingPersistedAnchors[added.trackableId];
-                    added.GetComponent<SampleAnchor>().Persisted = true;
+                    if (added.TryGetComponent(out SampleAnchor sampleAnchor))
+                    {
+                        sampleAnchor.Name = name;
+                        sampleAnchor.Persisted = true;
+                        sampleAnchor.TrackingState = added.trackingState;
+                    }
```

Comment:
should we log some error at this else condition?

Reply:
Hmm, I'm not sure. @KBonds, what do you think? This feels like a valid case if you choose to use pieces of the sample in your own app, but the script is pretty specific to our sample to begin with...